floor_generator/lib/processor/view_processor.dart

```diff
@@ -46,7 +46,11 @@ class ViewProcessor extends QueryableProcessor<View> {
         .getField(AnnotationField.viewQuery)
         ?.toStringValue();
 
-    if (query == null || !query.trimLeft().toLowerCase().startsWith('select')) {
+    final lowerQuery = query.toLowerCase();
+    if (query == null ||
+        (!lowerQuery.trimLeft().startsWith('select') &&
+            !(lowerQuery.trimLeft().startsWith('with') &&
+                lowerQuery.contains('select')))) {
```

Comment:
💡 We could think about extracting these checks into functions similar to shown in the following snippet. I realized that a `WITH` query needs **at least two** `SELECT`s. One in the `WITH` statement and another one in the statement that provides the database view.

```suggestion
    if (query == null || !(query.isSelectQuery || query.isCteWithSelect)) {
      throw _processorError.missingQuery;
    }
    return query;
  }
}

extension on String {
  bool get isSelectQuery => toLowerCase().trimLeft().startsWith('select');

  /// whether the string is a common table expression
  /// followed by a `SELECT` query
  bool get isCteWithSelect {
    final lowerCasedString = toLowerCase();
    return lowerCasedString.trimLeft().startsWith('with') &&
        'select'.allMatches(lowerCasedString).length >= 2;
  }
}
```